### ISSUE TITLE
llm训练loss相关问题

### ISSUE THREAD

**Issue by Maokui-He** [FunAudioLLM/CosyVoice#682]
你好，请问方便提供一些llm训练时候的loss信息吗，大概能到多少呢

**Comment by zhangyike** [comment]
请问50hz/25Hz模型的训练精度（acc）最终分别能到多少？学习率和warmup step是论文中写的1e-4和10000吗？

**Comment by 4ndroidev** [comment]
交流一下，你是从头开始训练吗？我找了一个数据集，从头开始训练，训了32轮，loss 到 1.5 左右，acc 到 0.21 左右，还没有实际投入到 tts 推理来，感觉 loss 有点高，acc 有点低。。也想请教一下官方 loss 下降经验，少走弯路。

**Comment by Maokui-He** [comment]
我没有从头训练用作者提供的llm初始化的，单轮训完loss在1.6左右，看上去下降幅度很小了，对了你的grad_norm大概多少，我的是0.3左右，感觉有点小

**Comment by ScottishFold007** [comment]
> 交流一下，你是从头开始训练吗？我找了一个数据集，从头开始训练，训了32轮，loss 到 1.5 左右，acc 到 0.21 左右，还没有实际投入到 tts 推理来，感觉 loss 有点高，acc 有点低。。也想请教一下官方 loss 下降经验，少走弯路。
> 
> learning_rate: 0.001，其他参数默认没改。
> 
> <img alt="image" width="998" src="https://private-user-images.githubusercontent.com/27715835/390305544-300c028f-cdb4-4d3f-ae9a-085e208799f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI2OTQ1OTEsIm5iZiI6MTczMjY5NDI5MSwicGF0aCI6Ii8yNzcxNTgzNS8zOTAzMDU1NDQtMzAwYzAyOGYtY2RiNC00ZDNmLWFlOWEtMDg1ZTIwODc5OWY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI3VDA3NTgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5ODViOTM4OTIwMmNkOGQ1NjRmNDlhNDYxOTYxODEyNmMwNjM4OTI4ZjM4MGY3NzlkZjQxMWVjYzI1OGE4YWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wZGCVtuBVX_kA_Lakmk-A9pypZSWSexWxIlchcMmjNg">

你这个数据量多少？有没有听实际的发声？数据少看loss、acc这些指标没什么用

**Comment by 4ndroidev** [comment]
> > 交流一下，你是从头开始训练吗？我找了一个数据集，从头开始训练，训了32轮，loss 到 1.5 左右，acc 到 0.21 左右，还没有实际投入到 tts 推理来，感觉 loss 有点高，acc 有点低。。也想请教一下官方 loss 下降经验，少走弯路。
> > learning_rate: 0.001，其他参数默认没改。
> > <img alt="image" width="998" src="https://private-user-images.githubusercontent.com/27715835/390305544-300c028f-cdb4-4d3f-ae9a-085e208799f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI2OTQ1OTEsIm5iZiI6MTczMjY5NDI5MSwicGF0aCI6Ii8yNzcxNTgzNS8zOTAzMDU1NDQtMzAwYzAyOGYtY2RiNC00ZDNmLWFlOWEtMDg1ZTIwODc5OWY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI3VDA3NTgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5ODViOTM4OTIwMmNkOGQ1NjRmNDlhNDYxOTYxODEyNmMwNjM4OTI4ZjM4MGY3NzlkZjQxMWVjYzI1OGE4YWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wZGCVtuBVX_kA_Lakmk-A9pypZSWSexWxIlchcMmjNg">
> 
> 你这个数据量多少？有没有听实际的发声？数据少看loss、acc这些指标没什么用

600 万个数据样本

**Comment by 4ndroidev** [comment]
> grad_norm

我把 log_per_step 这个方法删了...不想看一大堆日志...

**Comment by Maokui-He** [comment]
我数据量大概在900w条，训了一轮过后训练集内数据解码有片段能听到对应的音频，请问你训练的时候出除合成文本输入，还有其他的输入吗（e.g. prompt text、prompt audio、speaker embedding）
我目前只用了合成文本和speaker embedding

**Comment by 4ndroidev** [comment]
> 合成文本

就这些够了，看代码

**Comment by ScottishFold007** [comment]
> > grad_norm
> 
> 我把 log_per_step 这个方法删了...不想看一大堆日志...

试试wandb监控，不用默认的log：
<img width="1357" alt="image" src="https://github.com/user-attachments/assets/738489a9-09a5-4333-a4b4-5b45b41dc87d">
